## 毕业项目

分析数据商店信息

In [6]:
import pandas as pd
import numpy as np

data_store = pd.read_csv('./data/store.csv')
data_store.head(10)

Store StoreType Assortment  CompetitionDistance  CompetitionOpenSinceMonth  \
0      1         c          a               1270.0                        9.0   
1      2         a          a                570.0                       11.0   
2      3         a          a              14130.0                       12.0   
3      4         c          c                620.0                        9.0   
4      5         a          a              29910.0                        4.0   
5      6         a          a                310.0                       12.0   
6      7         a          c              24000.0                        4.0   
7      8         a          a               7520.0                       10.0   
8      9         a          c               2030.0                        8.0   
9     10         a          a               3160.0                        9.0   

   CompetitionOpenSinceYear  Promo2  Promo2SinceWeek  Promo2SinceYear  \
0                    2008.0       0              NaN              NaN   
1                    2007.0       1             13.0           2010.0   
2                    2006.0       1             14.0           2011.0   
3                    2009.0       0              NaN              NaN   
4                    2015.0       0              NaN              NaN   
5                    2013.0       0              NaN              NaN   
6                    2013.0       0              NaN              NaN   
7                    2014.0       0              NaN              NaN   
8                    2000.0       0              NaN              NaN   
9                    2009.0       0              NaN              NaN   

     PromoInterval  
0              NaN  
1  Jan,Apr,Jul,Oct  
2  Jan,Apr,Jul,Oct  
3              NaN  
4              NaN  
5              NaN  
6              NaN  
7              NaN  
8              NaN  
9              NaN

In [3]:
data_store.shape

(1115, 10)

In [16]:
data_store.describe()

Store  CompetitionDistance  CompetitionOpenSinceMonth  \
count  1115.00000          1112.000000                 761.000000   
mean    558.00000          5404.901079                   7.224704   
std     322.01708          7663.174720                   3.212348   
min       1.00000            20.000000                   1.000000   
25%     279.50000           717.500000                   4.000000   
50%     558.00000          2325.000000                   8.000000   
75%     836.50000          6882.500000                  10.000000   
max    1115.00000         75860.000000                  12.000000   

       CompetitionOpenSinceYear       Promo2  Promo2SinceWeek  Promo2SinceYear  
count                761.000000  1115.000000       571.000000       571.000000  
mean                2008.668857     0.512108        23.595447      2011.763573  
std                    6.195983     0.500078        14.141984         1.674935  
min                 1900.000000     0.000000         1.000000      2009.000000  
25%                 2006.000000     0.000000        13.000000      2011.000000  
50%                 2010.000000     1.000000        22.000000      2012.000000  
75%                 2013.000000     1.000000        37.000000      2013.000000  
max                 2015.000000     1.000000        50.000000      2015.000000

#### 数据总览

一共1115条数据，即1115家商店，共九个特征值。其中

- `StoreType` `Assortment` `CompetitionDistance`, `Promo2` 特征的数据较为完整，其他特征都有大量缺失值

#### 训练数据分析

In [32]:
train_data = pd.read_csv('./data/train.csv')
train_data.head(10)

Store  DayOfWeek        Date  Sales  Customers  Open  Promo StateHoliday  \
0      1          5  2015-07-31   5263        555     1      1            0   
1      2          5  2015-07-31   6064        625     1      1            0   
2      3          5  2015-07-31   8314        821     1      1            0   
3      4          5  2015-07-31  13995       1498     1      1            0   
4      5          5  2015-07-31   4822        559     1      1            0   
5      6          5  2015-07-31   5651        589     1      1            0   
6      7          5  2015-07-31  15344       1414     1      1            0   
7      8          5  2015-07-31   8492        833     1      1            0   
8      9          5  2015-07-31   8565        687     1      1            0   
9     10          5  2015-07-31   7185        681     1      1            0   

   SchoolHoliday  
0              1  
1              1  
2              1  
3              1  
4              1  
5              1  
6              1  
7              1  
8              1  
9              1

In [33]:
train_data.shape

(1017209, 9)

In [34]:
train_data = train_data.dropna(thresh=train_data.shape[0] * 0.5, axis = 1)
train_data.head(10)

Store  DayOfWeek        Date  Sales  Customers  Open  Promo StateHoliday  \
0      1          5  2015-07-31   5263        555     1      1            0   
1      2          5  2015-07-31   6064        625     1      1            0   
2      3          5  2015-07-31   8314        821     1      1            0   
3      4          5  2015-07-31  13995       1498     1      1            0   
4      5          5  2015-07-31   4822        559     1      1            0   
5      6          5  2015-07-31   5651        589     1      1            0   
6      7          5  2015-07-31  15344       1414     1      1            0   
7      8          5  2015-07-31   8492        833     1      1            0   
8      9          5  2015-07-31   8565        687     1      1            0   
9     10          5  2015-07-31   7185        681     1      1            0   

   SchoolHoliday  
0              1  
1              1  
2              1  
3              1  
4              1  
5              1  
6              1  
7              1  
8              1  
9              1

In [46]:
print(train_data['Open'].value_counts())

# 营业中
open_store = train_data[train_data['Open'] == 1]

print(open_store.count())

saled_store = open_store[open_store['Sales'] > 0]

print(saled_store.count())

data = pd.merge(data_store, saled_store, on='Store')
data.head()


1    844392
0    172817
Name: Open, dtype: int64
Store            844392
DayOfWeek        844392
Date             844392
Sales            844392
Customers        844392
Open             844392
Promo            844392
StateHoliday     844392
SchoolHoliday    844392
dtype: int64
Store            844338
DayOfWeek        844338
Date             844338
Sales            844338
Customers        844338
Open             844338
Promo            844338
StateHoliday     844338
SchoolHoliday    844338
dtype: int64


Store StoreType Assortment  CompetitionDistance  CompetitionOpenSinceMonth  \
0      1         c          a               1270.0                        9.0   
1      1         c          a               1270.0                        9.0   
2      1         c          a               1270.0                        9.0   
3      1         c          a               1270.0                        9.0   
4      1         c          a               1270.0                        9.0   

   CompetitionOpenSinceYear  Promo2  Promo2SinceWeek  Promo2SinceYear  \
0                    2008.0       0              NaN              NaN   
1                    2008.0       0              NaN              NaN   
2                    2008.0       0              NaN              NaN   
3                    2008.0       0              NaN              NaN   
4                    2008.0       0              NaN              NaN   

  PromoInterval  DayOfWeek        Date  Sales  Customers  Open  Promo  \
0           NaN          5  2015-07-31   5263        555     1      1   
1           NaN          4  2015-07-30   5020        546     1      1   
2           NaN          3  2015-07-29   4782        523     1      1   
3           NaN          2  2015-07-28   5011        560     1      1   
4           NaN          1  2015-07-27   6102        612     1      1   

  StateHoliday  SchoolHoliday  
0            0              1  
1            0              1  
2            0              1  
3            0              1  
4            0              1